core/engine/src/builtins/temporal/plain_date/mod.rs

```diff
@@ -958,47 +901,61 @@ pub(crate) fn to_partial_date_record(
     // TODO: Most likely need to use an iterator to handle.
     let day = partial_object
         .get(js_string!("day"), context)?
-        .map(|v| super::to_positive_integer_with_trunc(v, context))
+        .map(|v| {
+            let finite = v.to_finitef64(context)?;
+            finite
+                .as_positive_integer_with_truncation()
+                .map_err(JsError::from)
+        })
         .transpose()?;
     let month = partial_object
         .get(js_string!("month"), context)?
-        .map(|v| super::to_positive_integer_with_trunc(v, context))
+        .map(|v| {
+            let finite = v.to_finitef64(context)?;
+            finite
+                .as_positive_integer_with_truncation()
+                .map_err(JsError::from)
+        })
         .transpose()?;
     let month_code = partial_object
         .get(js_string!("monthCode"), context)?
         .map(|v| {
-            let JsValue::String(month_code) =
-                v.to_primitive(context, crate::value::PreferredType::String)?
-            else {
+            let v = v.to_primitive(context, crate::value::PreferredType::String)?;
+            let JsValue::String(month_code) = v else {
                 return Err(JsNativeError::typ()
                     .with_message("The monthCode field value must be a string.")
                     .into());
             };
-            TinyAsciiStr::<4>::from_str(&month_code.to_std_string_escaped())
+            TinyAsciiStr::<4>::try_from_str(&month_code.to_std_string_escaped())
                 .map_err(|e| JsError::from(JsNativeError::typ().with_message(e.to_string())))
         })
         .transpose()?;
     let year = partial_object
         .get(js_string!("year"), context)?
-        .map(|v| super::to_integer_if_integral(v, context))
+        .map(|v| {
+            let finite = v.to_finitef64(context)?;
+            Ok::<i32, JsError>(finite.as_integer_with_truncation::<i32>())
+        })
         .transpose()?;
     let era_year = partial_object
         .get(js_string!("eraYear"), context)?
-        .map(|v| super::to_integer_if_integral(v, context))
+        .map(|v| {
+            let finite = v.to_finitef64(context)?;
+            Ok::<i32, JsError>(finite.as_integer_with_truncation::<i32>())
+        })
         .transpose()?;
     let era = partial_object
         .get(js_string!("era"), context)?
         .map(|v| {
-            let JsValue::String(era) =
-                v.to_primitive(context, crate::value::PreferredType::String)?
-            else {
+            let v = v.to_primitive(context, crate::value::PreferredType::String)?;
+            let JsValue::String(era) = v else {
```

Comment:
Here and everywhere; we're moving to using `as_string()` instead of pattern matching on JsValue. Could you fix this? I've fixed `main` in my PR but I'd like to not have to play catch up with everyone.

See #4080